src/redux/preview.js

```diff
@@ -53,9 +57,28 @@ module.exports.previewReducer = (state, action) => {
         return Object.assign({}, state, {
             parent: action.info
         });
-    case 'SET_STUDIOS':
+    case 'SET_PROJECT_STUDIOS':
+        // also initialize currentStudioIds, to keep track of which studios
+        // the project is currently in.
         return Object.assign({}, state, {
-            studios: action.items
+            projectStudios: action.items,
+            currentStudioIds: action.items.reduce((ids, studio) => {
```

Comment:
Curious about the use of an object here rather than an array (or `Set`). Is it always `{<studioId1>: true, <studioId2>: true, ...}`?  The name `currentStudioIds` implies an array, so later on when you use the check `studioId in currentStudioIds`, it looks like a misuse of `in` on an array.

Reply:
Turns out that Sets are not supported as state types in Redux! (Trying to find a reference online now, but failing.) I implemented it with currentStudioIds as a Set, but redux was mysteriously replacing my nice, data-filled set with an empty object.

So yes, currentStudioIds is an object of form `{<studioId1>: true, <studioId2>: true, ...}`. Not an array. What might be a name that would make this clearer?